libraries/psio/include/psio/reflect.hpp

```diff
@@ -0,0 +1,760 @@
+
+   template <typename>
+   struct is_std_array : std::false_type
+   {
+   };
+
+   template <typename T, auto N>
+   struct is_std_array<std::array<T, N>> : std::true_type
+   {
+      using value_type                        = T;
+      constexpr static const std::size_t size = N;
+   };
+
+   template <typename T>
+   constexpr bool is_std_array_v = is_std_array<T>::value;
+
+   template <typename>
+   struct is_std_variant : std::false_type
+   {
+   };
+
+   template <typename... T>
+   struct is_std_variant<std::variant<T...>> : std::true_type
+   {
+      static std::string name() { return get_variant_typename<T...>(); }
+      template <typename First, typename... Rest>
+      static std::string get_variant_typename()
+      {
+         if constexpr (sizeof...(Rest) > 0)
+            return std::string(get_type_name<First>()) + "|" + get_variant_typename<Rest...>();
+         else
+            return std::string(get_type_name<First>()) + "|";
+      }
+      using alts_as_tuple                       = std::tuple<T...>;
+      constexpr static const uint16_t num_types = sizeof...(T);
+   };
+
+   template <typename T>
+   constexpr bool is_std_variant_v = is_std_variant<T>::value;
+
+   template <typename>
+   struct is_std_tuple : std::false_type
+   {
+   };
+
+   template <typename... T>
+   struct is_std_tuple<std::tuple<T...>> : std::true_type
+   {
+      static std::string name() { return get_tuple_typename<T...>(); }
+      template <typename First, typename... Rest>
+      static std::string get_tuple_typename()
+      {
+         if constexpr (sizeof...(Rest) > 0)
+            return std::string(get_type_name<First>()) + "&" + get_tuple_typename<Rest...>();
+         else
+            return std::string(get_type_name<First>()) + "&";
+      }
+   };
+
+   template <typename T>
+   constexpr bool is_std_tuple_v = is_std_tuple<T>::value;
+
+   template <typename>
+   struct is_std_map : std::false_type
+   {
+   };
+
+   template <typename K, typename V>
+   struct is_std_map<std::map<K, V>> : std::true_type
+   {
+      static const std::string& name()
+      {
+         static std::string n =
+             std::string("map<") + get_type_name<K>() + "," + get_type_name<V>() + ">";
+         return n;
+      }
+   };
+
+   template <typename T>
+   constexpr bool is_std_map_v = is_std_map<T>::value;
+
+   template <typename T>
+   struct remove_cvref
+   {
+      typedef std::remove_cv_t<std::remove_reference_t<T>> type;
+   };
+
+   template <typename T>
+   using remove_cvref_t = typename remove_cvref<T>::type;
+
+   template <typename... Ts>
+   struct TypeList
+   {
+      static constexpr int size = sizeof...(Ts);
+   };
+
+   template <typename... Ts>
+   std::tuple<Ts...> TupleFromTypeListImpl(TypeList<Ts...>&&);
+   template <typename T>
+   using TupleFromTypeList = decltype(TupleFromTypeListImpl(std::declval<T>()));
+
+   template <typename T>
+   struct MemberPtrType;
```

Comment:
This only seems to be used in psibase/common/include/serveGraphQL.hpp and that only uses it for SimplifiedArgumentTypes which is equivalent to `args_as_tuple`.